samcli/lib/providers/sam_function_provider.py

```diff
@@ -40,15 +40,17 @@ def __init__(self, template_dict, parameter_overrides=None):
         :param dict template_dict: SAM Template as a dictionary
         :param dict parameter_overrides: Optional dictionary of values for SAM template parameters that might want
             to get substituted within the template
+        :param bool ignore_code_extraction_warnings: Ignores Log warnings
         """
 
         self.template_dict = SamBaseProvider.get_template(template_dict, parameter_overrides)
+        self.ignore_code_extraction_warnings = ignore_code_extraction_warnings
```

Comment:
why save something in class variable when all we have is static methods and we keep passing them to every method? *Hint: Lets change static methods to class methods and use class variable* :)

Reply:
Yeah, i thought about this, this a dirty way to do this without changing a lot of code.

Reply:
@c2tarun This is historically how we have done things and the reason we did it this was was for testing. We test a lot of the private methods through out the codebase, and to make some of that easier we have `@staticmethod` on methods so we can pass all the values (expected and not expected) into the method. This allows us deeper control and testing over different parts of the codebase. We can change it but not sure the value out weighs the effort in doing this.